ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1450,3 +1459,374 @@ def test_ipa_filesystemspace_check(self, create_jumbo_file):
+            ]
+        )
+
+    def test_ipahealthcheck_certmongerca(self, getcert_ca):
+        """
+        Test that healthcheck detects that a certmonger-defined
+        CA is missing
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == "dogtag-ipa-ca-renew-agent":
+                assert check["result"] == "ERROR"
+                assert (
+                    check["kw"]["msg"]
+                    == "Certmonger CA '{key}' missing"
+                )
+
+    @pytest.fixture()
+    def rename_httpd_cert(self):
+        """
+        Fixture to rename http cert and revert the change.
+        """
+        self.master.run_command(
+            ["mv", paths.HTTPD_CERT_FILE, "%s.old" % paths.HTTPD_CERT_FILE]
+        )
+        yield
+        self.master.run_command(
+            ["mv", "%s.old" % paths.HTTPD_CERT_FILE, paths.HTTPD_CERT_FILE]
+        )
+
+    def test_ipahealthcheck_ipaopensslchainvalidation(self, rename_httpd_cert):
+        """
+        Test for IPAOpenSSLChainValidation when httpd cert is moved.
+        """
+        error_msg = "Can't open {} for reading".format(paths.HTTPD_CERT_FILE)
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
+                assert check["result"] == "ERROR"
+                assert error_msg in check["kw"]["reason"]
+
+    @pytest.fixture()
+    def rename_ipa_cert(self):
+        """
+        Fixture to rename ipa cert and revert the change.
+        """
+        self.master.run_command(
+            ["mv", paths.IPA_CA_CRT, "%s.old" % paths.IPA_CA_CRT]
+        )
+        yield
+        self.master.run_command(
+            ["mv", "%s.old" % paths.IPA_CA_CRT, paths.IPA_CA_CRT]
+        )
+
+    def test_opensslchainvalidation_ipa_ca_cert(self, rename_ipa_cert):
+        """
+        Test for IPAOpenSSLChainValidation when IPA CA cert is renamed
+        """
+        error_msg = "Error loading file {}\n".format(paths.IPA_CA_CRT)
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
+                assert check["result"] == "ERROR"
+                assert error_msg in check["kw"]["reason"]
+            elif check["kw"]["key"] == paths.RA_AGENT_PEM:
+                assert check["result"] == "ERROR"
+                assert error_msg in check["kw"]["reason"]
+
+    @pytest.fixture()
+    def replace_ipa_chain(self):
```

Comment:
Is this fixture used anywhere?

Reply:
It is supposed to be the fixture for test_opensslchainvalidation_ipa_ca_cert().

The fixture rename_ipa_cert() should be dropped or a separate test for this case added.

replace_ipa_chain() leaves the file there as readable but with an incomplete chain which better replicates what is seen in the user deployments. It is not likely that /etc/ipa/ca.crt will be removed or missing but it having the wrong chain is something we've seen.

Reply:
dropped rename_ipa_cert() fixture.